controllers/operatorconfig/operatorconfig_controller.go

```diff
@@ -108,13 +119,17 @@ func (r *Reconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Resu
 			}
 			csvToUpdate := copyToCast.(*olmv1alpha1.ClusterServiceVersion)
 			klog.Infof("Applying OperatorConfig: %s to Operator: %s via CSV: %s, %s", operator.OperatorConfig, operator.Name, csv.Name, csv.Namespace)
-			return r.configCsv(ctx, csvToUpdate, serviceConfig)
+			if err := r.configCsv(ctx, csvToUpdate, serviceConfig); err != nil {
+				klog.Errorf("Failed to apply OperatorConfig %s/%s to Operator: %s via CSV: %s, %s", registry.Namespace, operator.OperatorConfig, operator.Name, csv.Namespace, csv.Name)
+				return ctrl.Result{}, err
+			}
 		}
 	}
-	return ctrl.Result{}, nil
+	klog.Infof("Finished reconciling OperatorConfig for request: %s, %s", instance.Namespace, instance.Name)
+	return ctrl.Result{RequeueAfter: constant.DefaultSyncPeriod}, nil
```

Comment:
Why do we need to requeue if reconcile was successful?

Reply:
The main reason is to check if the changes made in CSV was reverted back to original state by OLM, or in the case that CSV was updated to a new version, so ODLM could have the chance to re-apply the settings every three hours.

The most ideal way is to watch CSVs proactively, but it would cause OOM possibly in All Namespace mode if we did not handle the cache properly.....

Reply:
Got it. Yea, we will need more complex way of handling the CSV caching in the next release.